### PR TITLE
Remove DeviceValue

### LINKED ISSUE #3290: Remove DeviceValue class
Left from #3154

There is no need of using CUBLAS_POINTER_MODE_DEVICE

### PR DESCRIPTION
**Pull request by PDoakORNL**
## Proposed changes
Closes #3290
This removes the DeviceValue class which only provided support for device side values for use a one cuBlas call.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Leconte

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- Yes This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
Start testing in-house

File: src/QMCWaveFunctions/Fermion/DiracMatrixComputeCUDA.hpp [hunk 1]
```diff
@@ -75,8 +74,8 @@ class DiracMatrixComputeCUDA : public Resource
   Vector<char, OffloadPinnedAllocator<char>> invM_ptrs_;
 
   // cuBLAS geam wants these.
-  DeviceValue<T_FP, OffloadPinnedAllocator<T_FP>> dev_one{1.0};
-  DeviceValue<T_FP, OffloadPinnedAllocator<T_FP>> dev_zero{0.0};
+  T_FP dev_one{1.0};
```
Comment (ye-luo) [inline src/QMCWaveFunctions/Fermion/DiracMatrixComputeCUDA.hpp:77]: Needs renaming. It is not device but host

File: src/QMCWaveFunctions/tests/test_DiracMatrixComputeCUDA.cpp [hunk 2]
```diff
@@ -53,18 +53,17 @@ TEST_CASE("DiracMatrixComputeCUDA_cuBLAS_geam_call", "[wavefunction][fermion]")
   OffloadPinnedMatrix<double> mat_c;
   mat_c.resize(n, n);
 
-  DeviceValue<double, OffloadPinnedAllocator<double>> dev_one(1.0);
-  DeviceValue<double, OffloadPinnedAllocator<double>> dev_zero(0.0);
+  double dev_one(1.0);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/test_DiracMatrixComputeCUDA.cpp:56]: Not dev_

File: src/QMCWaveFunctions/Fermion/DiracMatrixComputeCUDA.hpp (hunk 1 above)
Comment (PDoakORNL) [reply src/QMCWaveFunctions/Fermion/DiracMatrixComputeCUDA.hpp:77]: doh

**Review by ye-luo — APPROVED** [review]
LGTM

**Pull request merged**